### PR TITLE
Added root directory in file system

### PR DESCRIPTION
**Pull request by gapra-msft**
Resolves https://github.com/Azure/azure-sdk-for-java/issues/7247

### CODE REVIEW

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemAsyncClient.java [hunk 1]
```diff
@@ -73,6 +73,8 @@
 
     public static final String ROOT_FILESYSTEM_NAME = "$root";
 
+    public static final String ROOT_DIRECTORY_NAME = "";
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemAsyncClient.java]: Does this need to be part of the public API?

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemAsyncClient.java [hunk 2]
```diff
@@ -158,6 +160,22 @@ public DataLakeDirectoryAsyncClient getDirectoryAsyncClient(String directoryName
...
     }
 
+    /**
+     * Creates a new DataLakeDirectoryAsyncClient object by concatenating "" to the end of
+     * DataLakeFileSystemAsyncClient's URL. The new DataLakeDirectoryAsyncClient uses the same request policy pipeline
+     * as the DataLakeFileSystemAsyncClient.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileSystemAsyncClient.getRootDirectoryAsyncClient}
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemAsyncClient.java]: Does this code snippet exist?

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemClient.java [hunk 3]
```diff
@@ -114,6 +116,21 @@ public DataLakeDirectoryClient getDirectoryClient(String directoryName) {
...
     }
 
+    /**
+     * Initializes a new DataLakeDirectoryClient object by concatenating "" to the end of
+     * DataLakeFileSystemClient's URL. The new DataLakeDirectoryClient uses the same request policy pipeline as the
+     * DataLakeFileSystemClient.
+     *
+     * <p><strong>Code Samples</strong></p>
+     *
+     * {@codesnippet com.azure.storage.file.datalake.DataLakeFileSystemClient.getRootDirectoryClient}
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemClient.java]: Does this code snippet exist?

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemClient.java [hunk 4]
```diff
@@ -114,6 +116,21 @@ public DataLakeDirectoryClient getDirectoryClient(String directoryName) {
             blobContainerClient.getBlobClient(directoryName).getBlockBlobClient());
     }
 
+    /**
+     * Initializes a new DataLakeDirectoryClient object by concatenating "" to the end of
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemClient.java]: Just so it stylizes the `""` differently.

```suggestion
     * Initializes a new DataLakeDirectoryClient object by concatenating {@code ""} to the end of
```

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemAsyncClient.java [hunk 5]
```diff
@@ -158,6 +160,22 @@ public DataLakeDirectoryAsyncClient getDirectoryAsyncClient(String directoryName
             blockBlobAsyncClient);
     }
 
+    /**
+     * Creates a new DataLakeDirectoryAsyncClient object by concatenating "" to the end of
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemAsyncClient.java]: Just so it stylizes the `""` differently.

```suggestion
     * Initializes a new DataLakeDirectoryAsyncClient object by concatenating {@code ""} to the end of
```

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemAsyncClient.java (hunk 1 above)
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemAsyncClient.java]: If these are both staying in public API we should add Javadocs that explain their purpose.

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemAsyncClient.java (hunk 2 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemAsyncClient.java]: oh I think I forgot to add the code snippets

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemClient.java (hunk 3 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemClient.java]: Will add it

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemAsyncClient.java (hunk 1 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemAsyncClient.java]: It doesn't have to be. I'l change it to private.

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemClient.java (hunk 4 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemClient.java]: Done

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemAsyncClient.java (hunk 1 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileSystemAsyncClient.java]: I'll make them private

**Comment by gapra-msft** [comment]
@alzimmermsft Just addressed your comments!

**Pull request merged**